sql.go

```diff
@@ -3,21 +3,38 @@ package sql
 import (
 	dbsql "database/sql"
 	"fmt"
-
 	_ "github.com/denisenkom/go-mssqldb"
 	_ "github.com/go-sql-driver/mysql"
 	_ "github.com/lib/pq"
 	_ "github.com/mattn/go-sqlite3"
-
 	"go.k6.io/k6/js/modules"
 )
 
 func init() {
-	modules.Register("k6/x/sql", new(SQL))
+	modules.Register("k6/x/sql", new(ModuleInstance))
+}
+
+// ModuleInstance represents an instance of the SQL module for every VU.
```

Comment:
I believe that this isn't an instance that `represents an instance of the SQL module for every VU`. It's more like `RootModule` (e.g. ), which is initialized once per test run per test (and not sure if we need to embed the `*SQL` there). 

Check, for example, how this separation is implemented inside the k6's module:
https://github.com/grafana/k6/blob/master/js/modules/k6/html/html.go#L36-L50

Also, I'd recommend adding the lines, like:
```go
var (
	_ modules.Module   = &RootModule{}
	_ modules.Instance = &ModuleInstance{}
)
```

That will guarantee in the long term that the structs correctly implement interfaces.

Reply:
I definitely went through some iterations from the documentation and finding various examples, then trimming things, so some aspects may have blended incorrectly 😆. I'll look more at that example. It just seemed like a lot of that extra boilerplate code wasn't really applicable, so I didn't want to have it there to be confusing (it was to me anyway).

> and not sure if we need to embed the *SQL there

True. I'll remove that.

> 	_ modules.Module   = &RootModule{}
	_ modules.Instance = &ModuleInstance{}

I originally included this as the example from docs had this, but it kinda _felt_ like "code smell" in that these are unused variables. 🤷  I'll add it back in since this seems to be the recommended pattern.

Thanks for the great feedback! I'll get another update soon as I'll be making similar updates for some of the other extensions as well.

Reply:
For additional context to what my thought was, I trimmed things due to not really needing the access to the VU state from within extension. Since I did not need access, I felt that access to VU state from the script-side is best handled by the use of the `k6/exec` module. 

Do you think I should just retain that reference to the VU, still unused, for keeping usage consistent?

Reply:
> originally included this as the example from docs had this, but it kinda felt like "code smell" in that these are unused variables.

No worries, in that case, it's almost the semi-official way to prove that something is implementing the interface. You can read this as _go please try to compile or fail_.

>Do you think I should just retain that reference to the VU, still unused, for keeping usage consistent?

I can imagine the case when maybe we can use that. In the `func (*SQL) Query` we call `rows, err := db.Query(query, args...)`, but in theory, we can migrate to `rows, err := db.QueryContext(s.vu.Context(), query, args...)` (s is `*SQL`)

But on the other side not sure if this is important for us here. If we don't use the VU then I believe we don't need to include it.